### ISSUE TITLE
Should we have a preview length and truncate text?

### ISSUE THREAD

**Issue by readeral** [Vision100IT/pwad.org.au#7]
Multiple results with this much text could make for a lot of scrolling. Do we want a limited preview? I feel like you've mentioned this before and that this approach was a conscious decision but I can't remember...
<img width="1152" alt="Prayer of Approach 7" src="https://user-images.githubusercontent.com/15904136/73346437-f68d1900-42d9-11ea-9071-7f60ebb0ea43.png">

**Comment by BarryThePenguin** [comment]
It would be good to query sanity so it only returns the first n number of characters. I was doing this with keystone, but unsure how to get the same result with sanity

**Comment by readeral** [comment]
Yeah can be done, I'll dig it up

**Comment by readeral** [comment]
Oh right. All I did was pass the body portable text through this chain of functions and then truncate it after the data had been retrieved...

```
searchbody: page.body
            .map(item => item.children)
            .flat()
            .map(child => (child ? child.text : ''))
            .join(' ')
```

**Comment by readeral** [comment]
We could use this in our GROQ to reduce the amount of data we're requesting:

```
  query pageContent($page: String) {
    pageContentOne(filter: {id: $page}) {
      _id
      title
      content[0..1]
    }
  }
```

**Comment by readeral** [comment]
Oh, wrong query - go this one:

```
query textSearch($search: String) {
    textSearch(filter: {search: $search}) {
      ... on Document {
        _id
        _type
      }
      ... on Hymn {
        title
        content[0..1]
        keywords {
          _id
          name
        }
      }
      ... on Prayer {
        title
        content[0..1]
        keywords {
          _id
          name
        }
      }
      ... on Liturgy {
        title
        content[0..1]
        keywords {
          _id
          name
        }
      }
    }
  }
```

**Comment by readeral** [comment]
This has been addressed in #18
